homeassistant/components/husqvarna_automower_ble/config_flow.py

```diff
@@ -60,62 +63,190 @@ async def async_step_bluetooth(
         self.address = discovery_info.address
         await self.async_set_unique_id(self.address)
         self._abort_if_unique_id_configured()
-        return await self.async_step_confirm()
+        return await self.async_step_bluetooth_confirm()
 
-    async def async_step_confirm(
+    async def async_step_bluetooth_confirm(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
-        """Confirm discovery."""
-        assert self.address
+        """Confirm Bluetooth discovery."""
 
-        device = bluetooth.async_ble_device_from_address(
-            self.hass, self.address, connectable=True
+        if user_input is not None:
+            self.pin = user_input[CONF_PIN]
+            return await self.async_step_bluetooth_finalise()
+
+        return self.async_show_form(
+            step_id="bluetooth_confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_PIN): str,
```

Comment:
Don't we need to add a validator to ensure the given PIN is numerical?
Also, can we use the library here to validate the PIN?
Same comments for the user step.

Reply:
Any ideas how to do that? Just an example would be helpful